### ISSUE TITLE
Swagger editor no longer allows '$' in parameter names

### ISSUE THREAD

**Issue by korygin** [swagger-api/swagger-editor#2185]
### Q&A (please complete the following information)
 - OS: [e.g. macOS]
 - Browser: [e.g. chrome, safari]
 - Version: [e.g. 22]
 - Method of installation: [e.g. npm, dist assets]
 - Swagger-Editor version: [e.g. 3.10.0]
 - Swagger/OpenAPI version: [e.g. Swagger 2.0, OpenAPI 3.0]

### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
# your YAML here
 parameters:
    $count:
      name: $count
      in: query
      description: 'Include count of items, see [ODataCount](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374632)'
      schema:
        type: boolean
    $expand:
      name: $expand
      in: query
      description: 'Expand related entities, see [ODataExpand](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374621)'
      schema:
        type: string
    $filter:
      name: $filter
      in: query
      description: 'Filter items by property values, see [ODataFiltering](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374625)'
      schema:
        type: string
    $format:
      name: $format
      in: query
      description: 'Specifies the media type of response, see [ODataFormat](http://docs.oasis-open.org/odata/odata/v4.0/os/part1-protocol/odata-v4.0-os-part1-protocol.html#_Toc372793706)'
      schema:
        enum:
          - application/json;odata.metadata=none
          - application/json;odata.metadata=minimal
          - application/json;odata.metadata=full
        type: string
    $orderby:
      name: $orderby
      in: query
      description: 'Order items by property values, see [ODataSorting](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374629)'
      schema:
        type: string
    $select:
      name: $select
      in: query
      description: 'Select properties to be returned, see [ODataSelect](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374620)'
      schema:
        type: string
    $skip:
      name: $skip
      in: query
      description: 'Skip the first n items, see [ODataPaging-Skip](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374631)'
      schema:
        type: integer
    $top:
      name: $top
      in: query
      description: 'Show only the first n items, see [ODataPaging-Top](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374630)'
      schema:
        type: integer

Swagger-Editor configuration options:
```js
SwaggerEditor({
  // your config options here
})
```

```
?yourQueryStringConfig
```

### Describe the bug you're encountering
Swagger Editor shows "Component names can only contain the characters A-Z a-z 0-9 - . _" error for parameters that start with '$' character. When documenting OData RESTful APIs, all system query parameters start with '$', see [https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#_Toc31360955](url)

### Screenshots

![image](https://user-images.githubusercontent.com/14077250/85569917-f252f680-b600-11ea-821d-976c30420308.png)

**Comment by hkosova** [comment]
**TL;DR:** You CAN use `$` in parameter `name`, just not in the key names in the `components`.

<br/>

OpenAPI Specification [requires](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#components-object) that the component key names in the `components` section consist of the characters `A-Z a-z 0-9 - . _`. These key names are used in `$ref` paths:

```yaml
components:
  schemas:
    SchemaName:  # <--- #ref: '#/components/schemas/SchemaName'
      ...
  parameters:
    paramComponentName: # <--- '#/components/schemas/paramComponentName'
      name: myParam
      ...
```

But in case of parameters, the actual parameter name is specified by the `name` attribute and not by the component name. The component key name can be different from the parameter `name`. This means you can still use `$` in parameter names.

```yaml
parameters:
  - $ref: '#/components/parameters/count'

...

components:
  parameters:
    count:           # <--- no $ here, this is the key name is used in the $ref path
      name: $count   # <--- this is the actual parameter name!
      in: query
      description: 'Include count of items, see [ODataCount](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part1-protocol.html#_Toc445374632)'
      schema:
        type: boolean
```

**Comment by korygin** [comment]
OK, got it. Apologies for the false alarm, I'm going to close this now. Thanks!
